skfda/ml/regression/_fpca_regression.py

```diff
@@ -0,0 +1,150 @@
+from __future__ import annotations
+
+from typing import TypeVar
+
+import numpy as np
+from sklearn.utils.validation import check_is_fitted
+
+from ..._utils._sklearn_adapter import BaseEstimator, RegressorMixin
+from ...misc.regularization import L2Regularization
+from ...preprocessing.dim_reduction import FPCA
+from ...representation import FData
+from ...representation.basis import Basis, CustomBasis, FDataBasis
+from ._linear_regression import LinearRegression
+
+FPCARegressionSelf = TypeVar("FPCARegressionSelf", bound="FPCARegression")
+
+
+class FPCARegression(
+    BaseEstimator,
+    RegressorMixin,
+):
+    r"""Regression using Functional Principal Components Analysis.
+
+    It performs Functional Principal Components Analysis to reduce the
+    dimension of the functional data, and then uses a linear regression model
+    to relate the transformed data to a scalar value.
+
+    Args:
+        n_components: Number of principal components to keep.
+        intercept: If True, the linear model is calculated with an intercept.
+            Defaults to ``True``.
+        pca_regularization: Regularization parameter for the principal
+            component extraction. If None then no regularization is applied.
+            Defaults to ``None``.
+        regression_regularization: Regularization parameter for the linear
+            regression. If None then no regularization is applied.
+            Defaults to ``None``.
+        components_basis: Basis used for the principal components. If None
+            then the basis of the input data is used. Defaults to None.
+            It is only used if the input data is a FDataBasis object.
+
+    Attributes:
+        n\_components\_: Number of principal components used.
+        components\_: Principal components.
+        cofficents\_: Coefficients of the linear regression model.
```

Comment:
There is a typo here.
```suggestion
        coefficients\_: Coefficients of the linear regression model.
```